### PR TITLE
AspNetCore/Http Instrumentation to leverage native Activity Status.

### PR DESCRIPTION
**Pull request by Yun-Ting**
Related to https://github.com/open-telemetry/opentelemetry-dotnet/issues/3118

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs [hunk 1]
```diff
@@ -200,19 +200,17 @@ public override void OnStopActivity(Activity activity, object payload)
 
                 var response = context.Response;
 
-                activity.SetTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode);
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs:203]: We need to keep this.

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs [hunk 2]
```diff
@@ -91,11 +91,7 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Single(exportedItems);
             var activity = exportedItems[0];
 
-            Assert.Equal(200, activity.GetTagValue(SemanticConventions.AttributeHttpStatusCode));
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:94]: same here. keep this assert.

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/InProcServerTests.cs [hunk 3]
```diff
@@ -74,7 +74,8 @@ public async void ExampleTest()
             Assert.Equal("localhost:5000", activity.GetTagValue(SemanticConventions.AttributeHttpHost));
             Assert.Equal("GET", activity.GetTagValue(SemanticConventions.AttributeHttpMethod));
             Assert.Equal("1.1", activity.GetTagValue(SemanticConventions.AttributeHttpFlavor));
-            Assert.Equal(200, activity.GetTagValue(SemanticConventions.AttributeHttpStatusCode));
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/InProcServerTests.cs:77]: keep this.

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs [hunk 4]
```diff
@@ -583,8 +583,8 @@ public async Task TestCanceledRequest(string method)
             Assert.True(eventRecords.Records.TryDequeue(out KeyValuePair<string, Activity> exceptionEvent));
             Assert.Equal("Stop", exceptionEvent.Key);
 
-            Assert.NotNull(exceptionEvent.Value.GetTagValue(SpanAttributeConstants.StatusCodeKey));
-            Assert.Null(exceptionEvent.Value.GetTagValue(SpanAttributeConstants.StatusDescriptionKey));
+            Assert.True(exceptionEvent.Value.Status != ActivityStatusCode.Unset);
+            Assert.True(exceptionEvent.Value.StatusDescription == null);
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs:587]: nit: Assert.Null?

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs [hunk 5]
```diff
@@ -116,26 +116,25 @@ public IncomingRequestsCollectionsIsAccordingToTheSpecTests(WebApplicationFactor
             Assert.Equal("http", activity.GetTagValue(SemanticConventions.AttributeHttpScheme));
             Assert.Equal(urlPath, activity.GetTagValue(SemanticConventions.AttributeHttpTarget));
             Assert.Equal($"http://localhost{urlPath}{query}", activity.GetTagValue(SemanticConventions.AttributeHttpUrl));
-            Assert.Equal(statusCode, activity.GetTagValue(SemanticConventions.AttributeHttpStatusCode));
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs:119]: same as above.

File: src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs (hunk 1 above)
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs:203]: why is this removed?

Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs]: I revisited the spec and understood that both Span Status and the common attributes (http.status_code) should be set.
I've added the common attribute back. Thanks!

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs [hunk 6]
```diff
@@ -544,8 +544,8 @@ public async Task TestRequestWithException(string method)
             Assert.True(eventRecords.Records.TryDequeue(out KeyValuePair<string, Activity> exceptionEvent));
             Assert.Equal("Stop", exceptionEvent.Key);
 
-            Assert.NotNull(activity.GetTagValue(SpanAttributeConstants.StatusCodeKey));
-            Assert.NotNull(activity.GetTagValue(SpanAttributeConstants.StatusDescriptionKey));
+            Assert.True(exceptionEvent.Value.Status != ActivityStatusCode.Unset);
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs:547]: Assert on activity status and statusdescription

**Review by cijothomas — APPROVED** [review]
please add changelog entry too

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs (hunk 6 above)
Comment (Yun-Ting) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs]: Vishwesh and I chatted offline and we confirmed that the in this case, activity and the exceptionEvent are referencing the same object.
I've updated the assertions to be using activity to be consistent with the existing code anyhow.

**Comment by Yun-Ting** [comment]
> please add changelog entry too

Updated http and aspnetcore instrumentation changelogs.

**Review by vishweshbankwar — APPROVED** [review]
